community/ndp/packstream-v1/src/main/java/org/neo4j/packstream/PackStream.java

```diff
@@ -163,6 +163,28 @@ private PackStream()
     {
     }
 
+    /**
+     * Implemented by application-specific structures.
+     */
+    public interface StructType
```

Comment:
Similar to above - in the domain level type system, it's not just structs that are "special" or "different", PackStream has a wide range of ways to represent numbers, for instance, while the Neo4j type system only has two (float and integer). Eg; It perhaps makes sense to have the 'type' representation only talk about domain-level types, without differentiation for structs and types based directly on primitives.